openfisca_core/taxscales/single_amount_tax_scale.py

```diff
@@ -22,3 +23,23 @@ def calc(
         bracket_indices = numpy.digitize(tax_base, guarded_thresholds, right = right)
         guarded_amounts = numpy.array([0] + self.amounts + [0])
         return guarded_amounts[bracket_indices - 1]
+
+    def to_average(self) -> taxscales.LinearAverageRateTaxScale:
+        """ Converts this scale to a LinearAverageRateTaxScale.
+            Parameter keys change from ('threshold', 'amount') to ('threshold', 'rate').
+
+            Usage notes:
+             - Allows linear interpolation between point on the SingleAmountTaxScale using:
+                `interpolated = self.to_average().calc(tax_base)/tax_base` # must handle ZeroDivisionError
+        """
+        average_tax_scale = taxscales.LinearAverageRateTaxScale(
+            name = self.name,
+            option = self.option,
+            unit = self.unit,
+            )
+
+        if self.thresholds:
+            for threshold, amount in list(zip(self.thresholds, self.amounts)):
+                average_tax_scale.add_bracket(threshold, amount)
```

Comment:
You will have to compute the rates here Delta amounts / Delta threshold.  
You will have to deal carefully with the first bracket when the amount is zero.